### PR TITLE
[WIP] Added version to the v4 pester object #1501

### PR DESCRIPTION
**Pull request by Stephanevg**
## 1. General summary of the pull request
This PR adds the version to the v4 pester object.
Currently, the version is set to 4.10.2 which would be logical next version number of pester.

### CODE REVIEW

File: Functions/PesterState.ps1 [hunk 1]
```diff
@@ -8,7 +8,8 @@ function New-PesterState {
         [Pester.OutputTypes]$Show = 'All',
         [object]$PesterOption,
         [Switch]$RunningViaInvokePester,
-        [Hashtable[]] $ScriptBlockFilter
+        [Hashtable[]] $ScriptBlockFilter,
+        [version]$Version = "4.10.1"
```
[the remark this answers is not in the document]
Comment (Stephanevg) [reply Functions/PesterState.ps1]: Fixed in this commit 45c0fff75c95f49666c56fb1558e88be6a050f11

File: Functions/PesterState.ps1 [hunk 2]
```diff
@@ -36,15 +37,16 @@ function New-PesterState {
...
+            [version]$Version
         )
 
         #public read-only
         $TagFilter = $_tagFilter
         $ExcludeTagFilter = $_excludeTagFilter
         $TestNameFilter = $_testNameFilter
 
-
+        $script:version = $Version
```
[the remark this answers is not in the document]
Comment (Stephanevg) [reply Functions/PesterState.ps1]: Done -> 45c0fff75c95f49666c56fb1558e88be6a050f11

**Comment by Stephanevg** [comment]
@nohwnd  I modified as requested

File: Functions/PesterState.ps1 [hunk 3]
```diff
@@ -27,7 +28,7 @@ function New-PesterState {
         }
     }
 
-    & $SafeCommands['New-Module'] -Name PesterState -AsCustomObject -ArgumentList $TagFilter, $ExcludeTagFilter, $TestNameFilter, $SessionState, $Strict, $Show, $PesterOption, $RunningViaInvokePester -ScriptBlock {
+    & $SafeCommands['New-Module'] -Name PesterState -AsCustomObject -ArgumentList $TagFilter, $ExcludeTagFilter, $TestNameFilter, $SessionState, $Strict, $Show, $PesterOption, $RunningViaInvokePester, $version -ScriptBlock {
```
Comment (nohwnd) [inline Functions/PesterState.ps1:31]: use `$Version`

File: Pester.psm1 [hunk 4]
```diff
@@ -1180,7 +1180,7 @@ function Invoke-Pester {
         if ($PassThru) {
             # Remove all runtime properties like current* and Scope
             $properties = @(
-                "TagFilter", "ExcludeTagFilter", "TestNameFilter", "ScriptBlockFilter", "TotalCount", "PassedCount", "FailedCount", "SkippedCount", "PendingCount", 'InconclusiveCount', "Time", "TestResult"
+                "version", "TagFilter", "ExcludeTagFilter", "TestNameFilter", "ScriptBlockFilter", "TotalCount", "PassedCount", "FailedCount", "SkippedCount", "PendingCount", 'InconclusiveCount', "Time", "TestResult"
```
Comment (nohwnd) [inline Pester.psm1:1183]: use `Version`

**Pull request closed without merging**